File: backend/src/main/java/com/tribly/dto/trips/response/TripListResponse.java
package com.tribly.dto.trips.response;

import java.util.List;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

@Schema(description = "Paginated trip list response")
public record TripListResponse(
    @Schema(description = "List of trips", required = true) List<TripDto> trips,
    @Schema(description = "Total number of trips", required = true) long total,
    @Schema(description = "Current page number", required = true) int page,
    @Schema(description = "Page size", required = true) int size) {}

File: frontend/src/api/dto/tripListResponse.ts
import type { TripDto } from './tripDto.ts'

/**
 * Paginated trip list response
 */
export interface TripListResponse {
  /** List of trips */
  trips: TripDto[]
  /** Total number of trips */
  total: number
  /** Current page number */
  page: number
  /** Page size */
  size: number
}
